AbletonMCP_Remote_Script/__init__.py

```
@@ -1,1062 +1,670 @@
+        # --- Audio ---
+        elif cmd == "get_audio_clip_info":
+            return handlers.audio.get_audio_clip_info(song, p.get("track_index", 0), p.get("clip_index", 0), ctrl)
+        elif cmd == "analyze_audio_clip":
+            return handlers.audio.analyze_audio_clip(song, p.get("track_index", 0), p.get("clip_index", 0), ctrl)
+
+        # --- Arrangement ---
+        elif cmd == "get_arrangement_clips":
+            return handlers.arrangement.get_arrangement_clips(song, p.get("track_index", 0), ctrl)
+
+        else:
+            raise Exception("Unhandled read-only command: " + cmd)
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)